packages/command-utils/lib/index.js

```diff
@@ -42,93 +42,78 @@ function error(...args) {
 }
 
 class Command {
-  constructor(bin, args = [], vars = []) {
-    Object.defineProperties(this, {
-      vars: {
-        value: vars
-      },
-      bin: {
-        value: resolveBin(bin)
-      },
-      args: {
-        value: args
-      }
-    })
-  }
-  toString() {
-    return `${this.vars.length > 0 ? `${this.vars.join(' ')} ` : ''}${this.bin}`
-  }
-  get bin() {
-    return this.bin
-  }
-  get args() {
-    return this.args
-  }
-  get vars() {
-    return this.vars
+  /**
+   * the command binary, e.g. `myScript.js` or `pwd`
+   * @type {string}
+   */
+  bin
+  /**
+   * The arguments passed to the command
+   * @type {string[]}
+   */
+  args
+  /**
+   * Environment variables in the form of key-value pairs
+   * @type {Object.<string, string>}
+   */
+  envVars
+
+  /**
+   * @param bin {string}  The binary name, e.g. `pwd`
+   * @param args {string[]} command arguments
+   * @param envVars {Object.<string, string>} Environment variables
+   */
+  constructor(bin, args = [], envVars = {}) {
+    this.bin = resolveBin(bin)
+    this.args = args
+    this.envVars = envVars
   }
 }
 
-function getCommand(bin, args = [], vars = []) {
-  return new Command(bin, args, vars)
+/**
+ * @param bin {string}  The binary name, e.g. `pwd`
+ * @param args {string[]} command arguments
+ * @param envVars {Object.<string, string>} Environment variables
+ */
+function getCommand(bin, args, envVars) {
+  return new Command(bin, args, envVars)
 }
 
-function runCommandsConcurrently(commands) {
-  const args = [
-    '--kill-others-on-fail',
-    '--prefix',
-    '[{name}]',
-    '--names',
-    Object.keys(commands).join(','),
-    '--prefix-colors',
-    'bgBlue.bold,bgMagenta.bold,bgGreen.bold',
-    '--success',
-    'all'
-  ]
-
-  Object.keys(commands).forEach((name) => {
-    let commandList = commands[name]
-
-    if (commandList) {
-      commandList = Array.isArray(commandList) ? commandList : [commandList]
-      commandList.forEach((command) => {
-        args.push(
-          `${command.toString()}${
-            command.args.length > 0 ? ` ${command.args.join(' ')} ` : ''
-          }`
-        )
-      })
+/**
+ * Runs all the given commands concurrently
+ * @param {Object.<string, Command | Command[]>} commands Command objects.
+ *  Their environment variables will be passed to every executable.
+ * @returns Promise A Promise that completed if all the commands are completed
+ */
+async function runCommandsConcurrently(commands) {
+  const promises = []
+  for (const cl of Object.values(commands)) {
+    const commandList = Array.isArray(cl) ? cl : [cl]
+    for (const command of commandList) {
+      promises.push(runCommandAsync(command.bin, command.args, command.envVars))
     }
-  })
-
-  let result = { status: 1 }
-
-  try {
-    result = runCommandSync(`${resolveBin('concurrently')}`, args)
```

Comment:
This was the biggest issue. This is an ugly hack to allow synchronous parallel execution in node (= execute multiple scripts simultaneously synchronously). This was accomplished by running `concurrently` (a lib to execute multiple scripts at the same time) synchronously. This had multiple issues:
- we needed to pass arguments and env variables in a huge string like `concurrently "ENV_VAR1=A, ENV_VAR2=B scriptA --arg1 abc --arg2 bcd" ENV_VAR3=3 scriptB --arg1=123` that is very hard to read and is not Windows compatible. 
- We were running `concurrently` not with native code but with a lib called `execa`. Dontknow why
- All these scripts executing scripts made IDE debuggers get lost unable to place breakpoints
- some of them also mess with `stdout` causing formatting like colors to be lost.

This solution is pure native code, easy to read with a price that `runCommandsConcurrently` became async (we never made use of its synchronous nature)